enterprise/server/remote_execution/action_merger/action_merger.go

```diff
@@ -26,18 +27,26 @@ const (
 
 var (
 	enableActionMerging = flag.Bool("remote_execution.enable_action_merging", true, "If enabled, identical actions being executed concurrently are merged into a single execution.")
+	hedgedActionCount   = flag.Int("remote_execution.action_merging_hedge_count", 0, "When action merging is enabled, this flag controls how many additional, 'hedged' attempts an action is run in the background. Note that even hedged actions are run at most once per execution request.")
 )
 
-func redisKeyForPendingExecutionID(ctx context.Context, adResource *digest.ResourceName) (string, error) {
+// Returns the redis keys that store execution-keyed action merging state. The
+// first key points from action ID to the first execution ID that was started
+// for that action (which will usually finish first), and the second key points
+// from an action ID to the number of attempted executions of that action,
+// which is used to track hedging.
+func redisKeysForPendingExecutionID(ctx context.Context, adResource *digest.ResourceName) (string, string, error) {
 	userPrefix, err := prefix.UserPrefixFromContext(ctx)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 	downloadString, err := adResource.DownloadString()
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
-	return fmt.Sprintf("pendingExecution/%s%s", userPrefix, downloadString), nil
+	return fmt.Sprintf("pendingExecution/%s%s", userPrefix, downloadString),
```

Comment:
Instead of separating this information into two separate keys, why don't we create a new hash key that contains both pieces of information?
I was fine with having the second key until I saw the `strings.Replace` call.

Reply:
Good idea, done